src/kloewe/dot/src/dot.h

```diff
@@ -0,0 +1,92 @@
+/*----------------------------------------------------------------------------
+  File    : dot.h
+  Contents: dot product (cpu dispatcher)
+  Author  : Kristian Loewe
+----------------------------------------------------------------------------*/
+#ifndef DOT_H
+#define DOT_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/*----------------------------------------------------------------------------
+  Enum to encode the sets of dot implementations
+----------------------------------------------------------------------------*/
+enum flags {
+    DOT_NAIVE  = 1,
+    DOT_SSE2   = 2,
+    DOT_AVX    = 3,
+    #ifndef DOT_NOFMA
+    DOT_AVXFMA = 4,
+    #endif
+    DOT_AUTO   = 3
```

Comment:
Currently, the values/variants are ordered chronologically wrt the advent of the prerequisite instruction set extensions. Usually, this order should coincide with better benchmark performance. How about we keep the order but still set `DOT_AUTO = 100`? The automatic choice can still be made to ignore DOT_AVXFMA unless its explicitly chosen. If the situation should change in the future, for example, if the avx-fma version can be improved somehow, only dot_set_impl() needs to be adapted.